### PR TITLE
Change aws kms input to require an arn string

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
This improve UX by removing the need to pre create a kube secret and require only a string as input instead. This align UX with RolesRef field.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Ref https://issues.redhat.com/browse/HOSTEDCP-700

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: api/v1alpha1/hostedcluster_types.go [hunk 1]
```diff
@@ -1623,10 +1623,55 @@ type AWSKMSSpec struct {
...
+	//				"kms:Decrypt",
+	//				"kms:ReEncrypt*",
+	//				"kms:GenerateDataKey*",
+	//				"kms:DescribeKey"
+	//			],
+	//			"Resource": %q
+	//		}
+	//	]
+	// }
+	AWSKMSRoleARN string `json:"awskmsRoleARN"`
```
Comment (derekwaynecarr) [inline api/v1alpha1/hostedcluster_types.go:1674]: awsKms?

File: api/v1beta1/hostedcluster_types.go [hunk 2]
```diff
@@ -1562,12 +1562,50 @@ type AWSKMSSpec struct {
...
+	//				"kms:Decrypt",
+	//				"kms:ReEncrypt*",
+	//				"kms:GenerateDataKey*",
+	//				"kms:DescribeKey"
+	//			],
+	//			"Resource": %q
+	//		}
+	//	]
+	// }
+	AWSKMSRoleARN string `json:"awskmsRoleARN"`
```
Comment (derekwaynecarr) [inline api/v1beta1/hostedcluster_types.go:1608]: should json be awsKms?

**Comment by enxebre** [comment]
thanks @derekwaynecarr, changed.

**Pull request merged**